src/NuGet.Core/NuGet.Configuration/Utility/XmlUtility.cs

```diff
@@ -27,6 +27,18 @@ internal static XDocument GetOrCreateDocument(XDocument content, string fullPath
             return CreateDocument(content, fullPath);
         }
 
+        internal static string GetEncodedXMLName(string name)
+        {
+            try
+            {
+                return XmlConvert.VerifyName(name);
```

Comment:
If the name is already something that is valid in XML we don't want to encode it again, because for example:

If you encode `'a name'` you get `'a_x0020_name', but then if you encode `'a_x0020_name'`, you would get something different, even though `'a_x0020_name'` is already a valid xml name.